src/utils/hooks/useKubevirtDataPod/useKubevirtDataPod.ts

```diff
@@ -0,0 +1,105 @@
+import { useEffect, useMemo, useState } from 'react';
+import useWebSocket from 'react-use-websocket';
+
+import { isEmpty } from '@kubevirt-utils/utils/utils';
+import {
+  consoleFetch,
+  K8sResourceCommon,
+  WatchK8sResource,
+} from '@openshift-console/dynamic-plugin-sdk';
+
+import useDeepCompareMemoize from '../useDeepCompareMemoize/useDeepCompareMemoize';
+
+import { constructURL, getResourceVersion, registerResourceVersion } from './utils/utils';
+import useKubevirtDataPodFilters from './useKubevirtDataPodFilters';
+
+type UseKubevirtDataPod = <T extends K8sResourceCommon>(
```

Comment:
can you use `IoK8sApiCoreV1Pod` instead of `K8sResourceCommon`?

Reply:
well i looked on IoK8sApiCoreV1Pod , its a completely different animal. why do u think it can be replaced?

Reply:
`IoK8sApiCoreV1Pod` is the type for the Pod resource. But this hook does not return the Pod information. It uses the dedicated Pods to have filtered, paginated, and sorted resources like VMs.

Reply:
I thought we are creating a proxy pod so you can you Kubernetes API type, but I might be wrong :)